src/main/java/com/zerocracy/radars/viber/TkViber.java

```
@@ -30,20 +42,79 @@
  */
 public final class TkViber implements Take {
 
+    /**
+     * Farm to use.
+     */
+    private final Farm farm;
+
+    /**
+     * Reactions.
+     */
+    private final Reaction reaction;
+
+    /**
+     * Constructor.
+     * @param farm Farm to use
+     */
+    public TkViber(final Farm farm) {
+        this(farm, new ReProfile());
+    }
+
+    /**
+     * Constructor.
+     * @param farm Farm to use
+     * @param react Reaction to use
+     */
+    TkViber(final Farm farm, final Reaction react) {
+        this.farm = farm;
+        this.reaction = react;
+    }
+
     @Override
-    public Response act(final Request request) throws IOException {
+    public Response act(final Request req) throws IOException {
+        final JsonObject json = TkViber.json(
+            new InputStreamReader(req.body(), StandardCharsets.UTF_8)
+        );
+        if (json.isEmpty()) {
+            throw new RsForward(
+                new RsParFlash(
+                    new Par(
+                        "We expect this URL to be called by Viber",
+                        "with JSON as body of the request"
+                    ).say(),
+                    Level.WARNING
+                )
+            );
+        }
+        if (Objects.equals(json.getString("event"), "message")) {
+            this.reaction.react(
+                new VbBot(), this.farm, new VbEvent.Simple(json)
+            );
+        }
         return new RsWithStatus(
-            new RsWithBody("Not yet implemented."),
-            HttpURLConnection.HTTP_NOT_IMPLEMENTED
+            HttpURLConnection.HTTP_OK
         );
     }
+
+    /**
+     * Read JSON from body.
+     * @param body The body
+     * @return The JSON object
+     */
+    private static JsonObject json(final Reader body) {
+        try (
+            final JsonReader reader = Json.createReader(body)
+        ) {
+            return reader.readObject();
+        } catch (final JsonException ex) {
```

@krzyk Is this not a runtime exception? I don't remember ever having to catch ``JsonException``.

---

@amihaiemil yes it is, but it is easier to debug if we see the whole json that failed, so I added it into the message

---

@krzyk hmm... I would personally let it fail and avoid this static method.  But let's see what the Kirill says :D

---

@krzyk I agree that it's better to rethrow more verbose exception here, but `String.format` won't work - at best case it prints something like this `"Can't parse JSON: java.io.StringReader@1234123"`